python/lsst/analysis/tools/tasks/makeMetricTable.py

```diff
@@ -28,81 +28,94 @@
 import lsst.pipe.base as pipeBase
 import numpy as np
 from astropy.table import Table
-from lsst.pex.config import Field
+from lsst.pex.config import ListField
 from lsst.pipe.base import connectionTypes as ct
 from lsst.skymap import BaseSkyMap
 
 
 class MakeMetricTableConnections(
     pipeBase.PipelineTaskConnections,
-    dimensions=("skymap",),
-    defaultTemplates={"metricBundleName": "objectTableCore_metrics"},
+    dimensions=(),
+    defaultTemplates={"metricBundleName": ""},
 ):
     data = ct.Input(
         doc="Metric bundle to read from the butler",
         name="{metricBundleName}",
         storageClass="MetricMeasurementBundle",
         deferLoad=True,
-        dimensions=(
-            "skymap",
-            "tract",
-        ),
+        dimensions=(),
         multiple=True,
     )
 
-    skymap = ct.Input(
-        doc="The skymap that covers the tract that the data is from.",
-        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
-        storageClass="SkyMap",
-        dimensions=("skymap",),
-    )
-
     metricTable = ct.Output(
         doc="A summary table of all metrics by tract.",
         name="{metricBundleName}Table",
         storageClass="ArrowAstropy",
-        dimensions=("skymap",),
+        dimensions=(),
     )
 
     def __init__(self, *, config=None):
         super().__init__(config=config)
-        if config.loadMatchedVisitData:
-            self.data = ct.Input(
-                doc="Metric bundle to read from the butler.",
-                name=config.connections.metricBundleName,
-                storageClass="MetricMeasurementBundle",
-                deferLoad=True,
-                dimensions=(
-                    "instrument",
-                    "skymap",
-                    "tract",
-                ),
-                multiple=True,
+        self.data = ct.Input(
+            doc=self.data.doc,
+            name=self.data.name,
+            storageClass=self.data.storageClass,
+            deferLoad=self.data.deferLoad,
+            dimensions=frozenset(sorted(config.inputDataDimensions)),
+            multiple=True,
+        )
+        self.metricTable = ct.Output(
+            doc=self.metricTable.doc,
+            name=self.metricTable.name,
+            storageClass=self.metricTable.storageClass,
+            dimensions=frozenset(sorted(config.outputDataDimensions)),
+        )
+        if "tract" in config.inputDataDimensions:
```

Comment:
I'm just flagging this as something to check during review, as I'm not entirely sure whether this is the best way to check if a skymap is needed.